From 057f74745ee652a00cd1ffb29bc2f7fa504a5a75 Mon Sep 17 00:00:00 2001
From: Sven Nierlein <sven@nierlein.de>
Date: Fri, 10 Nov 2017 16:46:18 +0100
Subject: [PATCH] make session file usage more robust

Compare the api returned url and the given url2connect less error prone.
Using --sslport=443 results in a url2connect https://vcenter:443/... while
get_option returns https://vcenter/...
Also it seems like some api returns .../sdk while others return .../sdk/webService
so just ignore that part.
Both leads to not resuing the existing session files.

this should fix #115
---
 check_vmware_esx.pl | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/check_vmware_esx.pl b/check_vmware_esx.pl
index 26739eb..e5841c0 100755
--- a/check_vmware_esx.pl
+++ b/check_vmware_esx.pl
@@ -1765,7 +1765,7 @@
       close (SESSION_LOCK_FILE);    
    
       eval {Vim::load_session(session_file => $sessionfile_name)};
-      if (($@ ne '') || (Opts::get_option("url") ne $url2connect))
+      if (($@ ne '') || (trim_connect_url(Opts::get_option("url")) ne trim_connect_url($url2connect)))
          {
          unlink $sessionfile_name;
          Util::connect($url2connect, $username, $password);
@@ -2296,6 +2296,14 @@ sub convert_number
     return $state;
     }
 
+sub trim_connect_url
+    {
+    my($url) = @_;
+    $url =~ s/:443//gmx;
+    $url =~ s/\/webService$//gmx;
+    return($url);
+    }
+
 sub check_health_state
     {
     my ($actual_state) = shift(@_);
